source/macos/AdaptiveCards/AdaptiveCards/Utils/ImageUtils.swift

```diff
@@ -62,4 +62,149 @@ class ImageUtils {
+        // This constraint need for image size [small, medium, large]
+        constraints.append(NSLayoutConstraint(item: imageView, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1.0, constant: cgSize.width))
+        constraints.append(NSLayoutConstraint(item: imageView, attribute: .height, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1.0, constant: cgSize.height))
+        constraints[0].priority = priority
+        constraints[0].identifier = ImageViewConstraint.imageWidth.rawValue
+        constraints[1].priority = priority
+        constraints[1].identifier = ImageViewConstraint.imageHeight.rawValue
+        
+        // This constraint fit image in container with a aspect ratio
+        let aspectRatio = ACRImageProperties.convertToAspectRatio(cgSize)
+        if aspectRatio.heightToWidth != 0 {
+            constraints.append(NSLayoutConstraint(item: imageView, attribute: .height, relatedBy: .equal, toItem: imageView, attribute: .width, multiplier: aspectRatio.heightToWidth, constant: 0))
+        } else if aspectRatio.widthToHeight != 0 {
+            constraints.append(NSLayoutConstraint(item: imageView, attribute: .width, relatedBy: .equal, toItem: imageView, attribute: .height, multiplier: aspectRatio.widthToHeight, constant: 0))
+        }
+        // Give the aspect ratio constraint a two-digit priority boost for first fulfilment. Priorities are calculated when the window loads the view. Otherwise, a constraint conflict will occur.
+        constraints[2].priority = priority + 2
+        constraints[2].identifier = ImageViewConstraint.imageAspect.rawValue
+        
+        if imageProperties.acsImageSize == .auto {
+            constraints.append(imageView.widthAnchor.constraint(lessThanOrEqualToConstant: imageProperties.contentSize.width))
+            constraints[3].identifier = ImageViewConstraint.imageAuto.rawValue
+        }
+        
+        // remove old constraint to avoid dublicates
+        for constraint in imageView.constraints {
+            if ImageViewConstraint(rawValue: constraint.identifier ?? "") != nil {
+                NSLayoutConstraint.deactivate([constraint])
+            }
+        }
+        NSLayoutConstraint.activate(constraints)
+        superView.update(imageProperties: imageProperties)
+    }
+    
+    // Prioritize layout based on content hugging.
+    static func getImageUILayoutPriority(_ wrapView: NSView?) -> NSLayoutConstraint.Priority {
+        if let wrapView = wrapView {
+            let priority = wrapView.contentHuggingPriority(for: .horizontal)
+            return (priority > .init(249)) ? .defaultHigh : priority
+        }
+        return .defaultHigh
+    }
+    
+    // return ACRImageWrappingView
+    static func returnImageWrappingView(element: ACSImage, hostConfig: ACSHostConfig, rootView: ACRView, parentView: NSView, isImageSet: Bool = false, imageSetImageSize: ACSImageSize = .none) -> ACRImageWrappingView {
```

Comment:
can we change name of this methods.
start with `return` not good name convention.

Reply:
updated.